Part 1)

Deep neural networks have proven, in my opinion, to be more than a bust as we have seen in the past. Whether it was writing music with Google’s Bach Doodle from March, or automatic driving cars from Tesla and Audi that have proven to be better than humans at avoiding collisions, save a few human caused examples. I think it is no question they have proven to integrate into our society and make meaningful change. The bigger question is what that means for humans.

When we create these machines, we add a value to them. Nothing we code or create is truly neutral. Because of this reformed view, the networks we design change the ecology of the system they are implemented in. Though I don’t think the advancements are a bubble, our reactions to them are more important. They will affect us in the future in the way we interact with the world. They will change who we meet socially (online or otherwise), what we see and consume in media, and how we deal with conflict in our borders. We are already seeing this change, and I cannot help but think that this will continue to become apart of our society for years to come. 


Part 2) Turned in a paper copy of my calculations, see the bin in front of Vander Linden's office.

Part 3)

In [1]:
"""
This module uses a Keras implementation to train a model on the fashion_mnist dataset

Written by Royce Lloyd for CS344 at Calvin College
4/12/2019
"""

from keras.datasets import fashion_mnist
from keras.utils import to_categorical
from keras import models
from keras import layers

(train_images, train_labels), (test_images, test_labels) = fashion_mnist.load_data()

train_images = train_images.reshape((60000, 28*28))
train_images = train_images.astype('float32') / 255

test_images = test_images.reshape((10000, 28*28))
test_images = test_images.astype('float32') / 255

train_labels = to_categorical(train_labels)
test_labels = to_categorical(test_labels)

model = models.Sequential()
model.add(layers.Dense(1086, activation='relu', input_shape=(28*28,)))
model.add(layers.Dense(10, activation='softmax'))
model.compile(optimizer='rmsprop', loss='binary_crossentropy', metrics=['accuracy'])

model.fit(train_images, train_labels, epochs=5, batch_size=128)

test_loss, test_acc = model.evaluate(test_images, test_labels)
print('Test Accuracy: ', test_acc)



Using TensorFlow backend.


Instructions for updating:
Colocations handled automatically by placer.


Instructions for updating:
Use tf.cast instead.


Epoch 1/5


  128/60000 [..............................] - ETA: 1:16 - loss: 0.3316 - acc: 0.9000

  640/60000 [..............................] - ETA: 20s - loss: 0.4620 - acc: 0.8822 

 1152/60000 [..............................] - ETA: 13s - loss: 0.3863 - acc: 0.8923

 1664/60000 [..............................] - ETA: 11s - loss: 0.3406 - acc: 0.9004

 2176/60000 [>.............................] - ETA: 9s - loss: 0.2973 - acc: 0.9086 

 2688/60000 [>.............................] - ETA: 9s - loss: 0.2667 - acc: 0.9156

 3200/60000 [>.............................] - ETA: 8s - loss: 0.2450 - acc: 0.9208

 3712/60000 [>.............................] - ETA: 8s - loss: 0.2335 - acc: 0.9242

 4352/60000 [=>............................] - ETA: 7s - loss: 0.2205 - acc: 0.9272



 4992/60000 [=>............................] - ETA: 7s - loss: 0.2099 - acc: 0.9294



 5504/60000 [=>............................] - ETA: 7s - loss: 0.2023 - acc: 0.9313



 6016/60000 [==>...........................] - ETA: 7s - loss: 0.1979 - acc: 0.9319

 6528/60000 [==>...........................] - ETA: 6s - loss: 0.1929 - acc: 0.9330

 7040/60000 [==>...........................] - ETA: 6s - loss: 0.1884 - acc: 0.9341

 7552/60000 [==>...........................] - ETA: 6s - loss: 0.1830 - acc: 0.9357

 8064/60000 [===>..........................] - ETA: 6s - loss: 0.1806 - acc: 0.9363

 8576/60000 [===>..........................] - ETA: 6s - loss: 0.1777 - acc: 0.9368

 9088/60000 [===>..........................] - ETA: 6s - loss: 0.1739 - acc: 0.9378

 9728/60000 [===>..........................] - ETA: 6s - loss: 0.1708 - acc: 0.9385

10368/60000 [====>.........................] - ETA: 5s - loss: 0.1673 - acc: 0.9394

11008/60000 [====>.........................] - ETA: 5s - loss: 0.1649 - acc: 0.9397

11648/60000 [====>.........................] - ETA: 5s - loss: 0.1617 - acc: 0.9409

12288/60000 [=====>........................] - ETA: 5s - loss: 0.1590 - acc: 0.9416

12928/60000 [=====>........................] - ETA: 5s - loss: 0.1563 - acc: 0.9424

13568/60000 [=====>........................] - ETA: 5s - loss: 0.1541 - acc: 0.9430

14208/60000 [======>.......................] - ETA: 5s - loss: 0.1516 - acc: 0.9437

14848/60000 [======>.......................] - ETA: 4s - loss: 0.1493 - acc: 0.9444

15488/60000 [======>.......................] - ETA: 4s - loss: 0.1469 - acc: 0.9451

16128/60000 [=======>......................] - ETA: 4s - loss: 0.1450 - acc: 0.9458

16768/60000 [=======>......................] - ETA: 4s - loss: 0.1430 - acc: 0.9465

17408/60000 [=======>......................] - ETA: 4s - loss: 0.1416 - acc: 0.9468



18048/60000 [========>.....................] - ETA: 4s - loss: 0.1402 - acc: 0.9472

18560/60000 [========>.....................] - ETA: 4s - loss: 0.1388 - acc: 0.9477

19200/60000 [========>.....................] - ETA: 4s - loss: 0.1378 - acc: 0.9480

19840/60000 [========>.....................] - ETA: 4s - loss: 0.1364 - acc: 0.9485

20480/60000 [=========>....................] - ETA: 4s - loss: 0.1351 - acc: 0.9489

21120/60000 [=========>....................] - ETA: 4s - loss: 0.1340 - acc: 0.9492

21760/60000 [=========>....................] - ETA: 3s - loss: 0.1330 - acc: 0.9495



22400/60000 [==========>...................] - ETA: 3s - loss: 0.1320 - acc: 0.9499

22912/60000 [==========>...................] - ETA: 3s - loss: 0.1310 - acc: 0.9502

23552/60000 [==========>...................] - ETA: 3s - loss: 0.1298 - acc: 0.9506

24192/60000 [===========>..................] - ETA: 3s - loss: 0.1290 - acc: 0.9507

24704/60000 [===========>..................] - ETA: 3s - loss: 0.1284 - acc: 0.9510

25344/60000 [===========>..................] - ETA: 3s - loss: 0.1273 - acc: 0.9514

25984/60000 [===========>..................] - ETA: 3s - loss: 0.1264 - acc: 0.9518

26624/60000 [============>.................] - ETA: 3s - loss: 0.1255 - acc: 0.9521

27264/60000 [============>.................] - ETA: 3s - loss: 0.1247 - acc: 0.9523

27904/60000 [============>.................] - ETA: 3s - loss: 0.1240 - acc: 0.9526

28544/60000 [=============>................] - ETA: 3s - loss: 0.1232 - acc: 0.9529

29056/60000 [=============>................] - ETA: 3s - loss: 0.1229 - acc: 0.9529

29696/60000 [=============>................] - ETA: 3s - loss: 0.1221 - acc: 0.9531

30336/60000 [==============>...............] - ETA: 2s - loss: 0.1214 - acc: 0.9534

30976/60000 [==============>...............] - ETA: 2s - loss: 0.1208 - acc: 0.9536

31616/60000 [==============>...............] - ETA: 2s - loss: 0.1200 - acc: 0.9538

32256/60000 [===============>..............] - ETA: 2s - loss: 0.1193 - acc: 0.9541



32896/60000 [===============>..............] - ETA: 2s - loss: 0.1187 - acc: 0.9543

33408/60000 [===============>..............] - ETA: 2s - loss: 0.1182 - acc: 0.9544

33920/60000 [===============>..............] - ETA: 2s - loss: 0.1177 - acc: 0.9546

34560/60000 [================>.............] - ETA: 2s - loss: 0.1169 - acc: 0.9549

35072/60000 [================>.............] - ETA: 2s - loss: 0.1164 - acc: 0.9550



35712/60000 [================>.............] - ETA: 2s - loss: 0.1158 - acc: 0.9553



36352/60000 [=================>............] - ETA: 2s - loss: 0.1151 - acc: 0.9555

36992/60000 [=================>............] - ETA: 2s - loss: 0.1145 - acc: 0.9557

37632/60000 [=================>............] - ETA: 2s - loss: 0.1141 - acc: 0.9558

38272/60000 [==================>...........] - ETA: 2s - loss: 0.1136 - acc: 0.9560

38912/60000 [==================>...........] - ETA: 2s - loss: 0.1132 - acc: 0.9562

39424/60000 [==================>...........] - ETA: 2s - loss: 0.1127 - acc: 0.9563

39936/60000 [==================>...........] - ETA: 1s - loss: 0.1123 - acc: 0.9564

40576/60000 [===================>..........] - ETA: 1s - loss: 0.1116 - acc: 0.9567

41216/60000 [===================>..........] - ETA: 1s - loss: 0.1109 - acc: 0.9569



41856/60000 [===================>..........] - ETA: 1s - loss: 0.1106 - acc: 0.9570

42496/60000 [====================>.........] - ETA: 1s - loss: 0.1101 - acc: 0.9572

43136/60000 [====================>.........] - ETA: 1s - loss: 0.1097 - acc: 0.9574

43776/60000 [====================>.........] - ETA: 1s - loss: 0.1093 - acc: 0.9575

44416/60000 [=====================>........] - ETA: 1s - loss: 0.1087 - acc: 0.9577

45056/60000 [=====================>........] - ETA: 1s - loss: 0.1082 - acc: 0.9579

45696/60000 [=====================>........] - ETA: 1s - loss: 0.1078 - acc: 0.9580

46336/60000 [======================>.......] - ETA: 1s - loss: 0.1073 - acc: 0.9582

46976/60000 [======================>.......] - ETA: 1s - loss: 0.1069 - acc: 0.9583

47616/60000 [======================>.......] - ETA: 1s - loss: 0.1066 - acc: 0.9584

48256/60000 [=======================>......] - ETA: 1s - loss: 0.1062 - acc: 0.9586

48896/60000 [=======================>......] - ETA: 1s - loss: 0.1056 - acc: 0.9588

49536/60000 [=======================>......] - ETA: 1s - loss: 0.1052 - acc: 0.9589

50176/60000 [========================>.....] - ETA: 0s - loss: 0.1048 - acc: 0.9591

50816/60000 [========================>.....] - ETA: 0s - loss: 0.1045 - acc: 0.9592

51456/60000 [========================>.....] - ETA: 0s - loss: 0.1041 - acc: 0.9594

52096/60000 [=========================>....] - ETA: 0s - loss: 0.1036 - acc: 0.9595

52736/60000 [=========================>....] - ETA: 0s - loss: 0.1033 - acc: 0.9596

53376/60000 [=========================>....] - ETA: 0s - loss: 0.1030 - acc: 0.9598

53888/60000 [=========================>....] - ETA: 0s - loss: 0.1027 - acc: 0.9599

54528/60000 [==========================>...] - ETA: 0s - loss: 0.1025 - acc: 0.9600

55168/60000 [==========================>...] - ETA: 0s - loss: 0.1022 - acc: 0.9601

55808/60000 [==========================>...] - ETA: 0s - loss: 0.1018 - acc: 0.9602

56448/60000 [===========================>..] - ETA: 0s - loss: 0.1015 - acc: 0.9603



57088/60000 [===========================>..] - ETA: 0s - loss: 0.1013 - acc: 0.9604

57600/60000 [===========================>..] - ETA: 0s - loss: 0.1010 - acc: 0.9605

58112/60000 [============================>.] - ETA: 0s - loss: 0.1008 - acc: 0.9606

58752/60000 [============================>.] - ETA: 0s - loss: 0.1004 - acc: 0.9607

59392/60000 [============================>.] - ETA: 0s - loss: 0.1002 - acc: 0.9608

59904/60000 [============================>.] - ETA: 0s - loss: 0.1000 - acc: 0.9608

60000/60000 [==============================] - 6s 97us/step - loss: 0.0999 - acc: 0.9609


Epoch 2/5
  128/60000 [..............................] - ETA: 6s - loss: 0.0830 - acc: 0.9641



  640/60000 [..............................] - ETA: 6s - loss: 0.0737 - acc: 0.9705



 1152/60000 [..............................] - ETA: 6s - loss: 0.0746 - acc: 0.9698



 1664/60000 [..............................] - ETA: 6s - loss: 0.0755 - acc: 0.9697

 2176/60000 [>.............................] - ETA: 5s - loss: 0.0743 - acc: 0.9700

 2688/60000 [>.............................] - ETA: 5s - loss: 0.0730 - acc: 0.9704

 3328/60000 [>.............................] - ETA: 5s - loss: 0.0730 - acc: 0.9709

 3968/60000 [>.............................] - ETA: 5s - loss: 0.0702 - acc: 0.9721

 4608/60000 [=>............................] - ETA: 5s - loss: 0.0719 - acc: 0.9715

 5248/60000 [=>............................] - ETA: 5s - loss: 0.0727 - acc: 0.9709

 5888/60000 [=>............................] - ETA: 5s - loss: 0.0729 - acc: 0.9712



 6528/60000 [==>...........................] - ETA: 5s - loss: 0.0728 - acc: 0.9712



 7168/60000 [==>...........................] - ETA: 5s - loss: 0.0725 - acc: 0.9712

 7808/60000 [==>...........................] - ETA: 5s - loss: 0.0720 - acc: 0.9713

 8448/60000 [===>..........................] - ETA: 4s - loss: 0.0719 - acc: 0.9713

 9088/60000 [===>..........................] - ETA: 4s - loss: 0.0720 - acc: 0.9712



 9728/60000 [===>..........................] - ETA: 4s - loss: 0.0716 - acc: 0.9714



10368/60000 [====>.........................] - ETA: 4s - loss: 0.0713 - acc: 0.9715

11008/60000 [====>.........................] - ETA: 4s - loss: 0.0719 - acc: 0.9712

11648/60000 [====>.........................] - ETA: 4s - loss: 0.0721 - acc: 0.9710

12288/60000 [=====>........................] - ETA: 4s - loss: 0.0722 - acc: 0.9710

12928/60000 [=====>........................] - ETA: 4s - loss: 0.0722 - acc: 0.9710

13568/60000 [=====>........................] - ETA: 4s - loss: 0.0725 - acc: 0.9708

14208/60000 [======>.......................] - ETA: 4s - loss: 0.0720 - acc: 0.9710

14848/60000 [======>.......................] - ETA: 4s - loss: 0.0719 - acc: 0.9709

15488/60000 [======>.......................] - ETA: 4s - loss: 0.0718 - acc: 0.9709

16128/60000 [=======>......................] - ETA: 4s - loss: 0.0723 - acc: 0.9708

16640/60000 [=======>......................] - ETA: 4s - loss: 0.0719 - acc: 0.9709

17280/60000 [=======>......................] - ETA: 4s - loss: 0.0719 - acc: 0.9710

17920/60000 [=======>......................] - ETA: 3s - loss: 0.0719 - acc: 0.9709

18560/60000 [========>.....................] - ETA: 3s - loss: 0.0721 - acc: 0.9708

19200/60000 [========>.....................] - ETA: 3s - loss: 0.0719 - acc: 0.9709

19840/60000 [========>.....................] - ETA: 3s - loss: 0.0714 - acc: 0.9710



20480/60000 [=========>....................] - ETA: 3s - loss: 0.0717 - acc: 0.9710

20992/60000 [=========>....................] - ETA: 3s - loss: 0.0716 - acc: 0.9711

21504/60000 [=========>....................] - ETA: 3s - loss: 0.0713 - acc: 0.9711

22016/60000 [==========>...................] - ETA: 3s - loss: 0.0711 - acc: 0.9712

22656/60000 [==========>...................] - ETA: 3s - loss: 0.0710 - acc: 0.9711

23296/60000 [==========>...................] - ETA: 3s - loss: 0.0708 - acc: 0.9712



23936/60000 [==========>...................] - ETA: 3s - loss: 0.0710 - acc: 0.9712

24576/60000 [===========>..................] - ETA: 3s - loss: 0.0713 - acc: 0.9711

25216/60000 [===========>..................] - ETA: 3s - loss: 0.0711 - acc: 0.9711

25856/60000 [===========>..................] - ETA: 3s - loss: 0.0711 - acc: 0.9711

26496/60000 [============>.................] - ETA: 3s - loss: 0.0711 - acc: 0.9711

27136/60000 [============>.................] - ETA: 3s - loss: 0.0709 - acc: 0.9712

27776/60000 [============>.................] - ETA: 3s - loss: 0.0706 - acc: 0.9714

28416/60000 [=============>................] - ETA: 2s - loss: 0.0704 - acc: 0.9714

29056/60000 [=============>................] - ETA: 2s - loss: 0.0703 - acc: 0.9714

29696/60000 [=============>................] - ETA: 2s - loss: 0.0701 - acc: 0.9715

30336/60000 [==============>...............] - ETA: 2s - loss: 0.0700 - acc: 0.9717

30976/60000 [==============>...............] - ETA: 2s - loss: 0.0701 - acc: 0.9716

31616/60000 [==============>...............] - ETA: 2s - loss: 0.0700 - acc: 0.9717

32256/60000 [===============>..............] - ETA: 2s - loss: 0.0700 - acc: 0.9717

32896/60000 [===============>..............] - ETA: 2s - loss: 0.0700 - acc: 0.9716



33536/60000 [===============>..............] - ETA: 2s - loss: 0.0699 - acc: 0.9717

34176/60000 [================>.............] - ETA: 2s - loss: 0.0697 - acc: 0.9718

34816/60000 [================>.............] - ETA: 2s - loss: 0.0697 - acc: 0.9718

35328/60000 [================>.............] - ETA: 2s - loss: 0.0699 - acc: 0.9717

35840/60000 [================>.............] - ETA: 2s - loss: 0.0700 - acc: 0.9717

36352/60000 [=================>............] - ETA: 2s - loss: 0.0699 - acc: 0.9717

36992/60000 [=================>............] - ETA: 2s - loss: 0.0698 - acc: 0.9717

37632/60000 [=================>............] - ETA: 2s - loss: 0.0698 - acc: 0.9718

38272/60000 [==================>...........] - ETA: 2s - loss: 0.0698 - acc: 0.9718



38912/60000 [==================>...........] - ETA: 1s - loss: 0.0696 - acc: 0.9718

39552/60000 [==================>...........] - ETA: 1s - loss: 0.0695 - acc: 0.9719

40192/60000 [===================>..........] - ETA: 1s - loss: 0.0695 - acc: 0.9718

40832/60000 [===================>..........] - ETA: 1s - loss: 0.0696 - acc: 0.9719

41472/60000 [===================>..........] - ETA: 1s - loss: 0.0695 - acc: 0.9719

42112/60000 [====================>.........] - ETA: 1s - loss: 0.0694 - acc: 0.9720

42752/60000 [====================>.........] - ETA: 1s - loss: 0.0693 - acc: 0.9720

43392/60000 [====================>.........] - ETA: 1s - loss: 0.0693 - acc: 0.9720

44032/60000 [=====================>........] - ETA: 1s - loss: 0.0691 - acc: 0.9720

44672/60000 [=====================>........] - ETA: 1s - loss: 0.0689 - acc: 0.9721

45312/60000 [=====================>........] - ETA: 1s - loss: 0.0687 - acc: 0.9722

45952/60000 [=====================>........] - ETA: 1s - loss: 0.0686 - acc: 0.9723

46464/60000 [======================>.......] - ETA: 1s - loss: 0.0684 - acc: 0.9723

47104/60000 [======================>.......] - ETA: 1s - loss: 0.0685 - acc: 0.9723

47744/60000 [======================>.......] - ETA: 1s - loss: 0.0685 - acc: 0.9723

48384/60000 [=======================>......] - ETA: 1s - loss: 0.0684 - acc: 0.9724

49024/60000 [=======================>......] - ETA: 1s - loss: 0.0681 - acc: 0.9725

49664/60000 [=======================>......] - ETA: 0s - loss: 0.0680 - acc: 0.9726

50304/60000 [========================>.....] - ETA: 0s - loss: 0.0681 - acc: 0.9725

50944/60000 [========================>.....] - ETA: 0s - loss: 0.0680 - acc: 0.9725

51584/60000 [========================>.....] - ETA: 0s - loss: 0.0679 - acc: 0.9726

52224/60000 [=========================>....] - ETA: 0s - loss: 0.0678 - acc: 0.9726

52864/60000 [=========================>....] - ETA: 0s - loss: 0.0678 - acc: 0.9726

53504/60000 [=========================>....] - ETA: 0s - loss: 0.0678 - acc: 0.9726

54144/60000 [==========================>...] - ETA: 0s - loss: 0.0677 - acc: 0.9726

54784/60000 [==========================>...] - ETA: 0s - loss: 0.0678 - acc: 0.9725

55424/60000 [==========================>...] - ETA: 0s - loss: 0.0677 - acc: 0.9726

56064/60000 [===========================>..] - ETA: 0s - loss: 0.0676 - acc: 0.9726

56704/60000 [===========================>..] - ETA: 0s - loss: 0.0677 - acc: 0.9726

57216/60000 [===========================>..] - ETA: 0s - loss: 0.0678 - acc: 0.9726

57856/60000 [===========================>..] - ETA: 0s - loss: 0.0677 - acc: 0.9727

58496/60000 [============================>.] - ETA: 0s - loss: 0.0676 - acc: 0.9727

59136/60000 [============================>.] - ETA: 0s - loss: 0.0675 - acc: 0.9727

59776/60000 [============================>.] - ETA: 0s - loss: 0.0675 - acc: 0.9728



60000/60000 [==============================] - 6s 92us/step - loss: 0.0675 - acc: 0.9728


Epoch 3/5
  128/60000 [..............................] - ETA: 6s - loss: 0.0655 - acc: 0.9734

  640/60000 [..............................] - ETA: 6s - loss: 0.0624 - acc: 0.9761

 1280/60000 [..............................] - ETA: 6s - loss: 0.0543 - acc: 0.9788



 1920/60000 [..............................] - ETA: 5s - loss: 0.0544 - acc: 0.9786

 2560/60000 [>.............................] - ETA: 5s - loss: 0.0564 - acc: 0.9776

 3200/60000 [>.............................] - ETA: 5s - loss: 0.0577 - acc: 0.9771

 3840/60000 [>.............................] - ETA: 5s - loss: 0.0589 - acc: 0.9762



 4480/60000 [=>............................] - ETA: 5s - loss: 0.0592 - acc: 0.9760

 5120/60000 [=>............................] - ETA: 5s - loss: 0.0584 - acc: 0.9765

 5760/60000 [=>............................] - ETA: 5s - loss: 0.0591 - acc: 0.9761

 6272/60000 [==>...........................] - ETA: 5s - loss: 0.0600 - acc: 0.9756

 6912/60000 [==>...........................] - ETA: 4s - loss: 0.0597 - acc: 0.9758

 7552/60000 [==>...........................] - ETA: 4s - loss: 0.0605 - acc: 0.9755

 8064/60000 [===>..........................] - ETA: 4s - loss: 0.0608 - acc: 0.9754

 8576/60000 [===>..........................] - ETA: 4s - loss: 0.0611 - acc: 0.9753

 9216/60000 [===>..........................] - ETA: 4s - loss: 0.0607 - acc: 0.9755

 9856/60000 [===>..........................] - ETA: 4s - loss: 0.0606 - acc: 0.9757

10496/60000 [====>.........................] - ETA: 4s - loss: 0.0605 - acc: 0.9757

11136/60000 [====>.........................] - ETA: 4s - loss: 0.0604 - acc: 0.9760

11776/60000 [====>.........................] - ETA: 4s - loss: 0.0606 - acc: 0.9759



12416/60000 [=====>........................] - ETA: 4s - loss: 0.0612 - acc: 0.9756

13056/60000 [=====>........................] - ETA: 4s - loss: 0.0613 - acc: 0.9754

13696/60000 [=====>........................] - ETA: 4s - loss: 0.0613 - acc: 0.9754

14336/60000 [======>.......................] - ETA: 4s - loss: 0.0613 - acc: 0.9755



14976/60000 [======>.......................] - ETA: 4s - loss: 0.0614 - acc: 0.9754

15616/60000 [======>.......................] - ETA: 4s - loss: 0.0610 - acc: 0.9756

16256/60000 [=======>......................] - ETA: 4s - loss: 0.0609 - acc: 0.9757

16896/60000 [=======>......................] - ETA: 4s - loss: 0.0609 - acc: 0.9756



17536/60000 [=======>......................] - ETA: 3s - loss: 0.0606 - acc: 0.9756

18048/60000 [========>.....................] - ETA: 3s - loss: 0.0610 - acc: 0.9755

18688/60000 [========>.....................] - ETA: 3s - loss: 0.0609 - acc: 0.9756

19200/60000 [========>.....................] - ETA: 3s - loss: 0.0611 - acc: 0.9756

19840/60000 [========>.....................] - ETA: 3s - loss: 0.0611 - acc: 0.9755

20480/60000 [=========>....................] - ETA: 3s - loss: 0.0614 - acc: 0.9753

21120/60000 [=========>....................] - ETA: 3s - loss: 0.0611 - acc: 0.9754

21760/60000 [=========>....................] - ETA: 3s - loss: 0.0609 - acc: 0.9755

22400/60000 [==========>...................] - ETA: 3s - loss: 0.0607 - acc: 0.9755

23040/60000 [==========>...................] - ETA: 3s - loss: 0.0607 - acc: 0.9756

23680/60000 [==========>...................] - ETA: 3s - loss: 0.0609 - acc: 0.9755

24320/60000 [===========>..................] - ETA: 3s - loss: 0.0606 - acc: 0.9757

24960/60000 [===========>..................] - ETA: 3s - loss: 0.0604 - acc: 0.9758

25600/60000 [===========>..................] - ETA: 3s - loss: 0.0602 - acc: 0.9759

26240/60000 [============>.................] - ETA: 3s - loss: 0.0601 - acc: 0.9759

26880/60000 [============>.................] - ETA: 3s - loss: 0.0602 - acc: 0.9758

27520/60000 [============>.................] - ETA: 2s - loss: 0.0601 - acc: 0.9759

28160/60000 [=============>................] - ETA: 2s - loss: 0.0603 - acc: 0.9758

28800/60000 [=============>................] - ETA: 2s - loss: 0.0604 - acc: 0.9758

29440/60000 [=============>................] - ETA: 2s - loss: 0.0602 - acc: 0.9759



30080/60000 [==============>...............] - ETA: 2s - loss: 0.0602 - acc: 0.9759



30720/60000 [==============>...............] - ETA: 2s - loss: 0.0601 - acc: 0.9759

31360/60000 [==============>...............] - ETA: 2s - loss: 0.0603 - acc: 0.9758

32000/60000 [===============>..............] - ETA: 2s - loss: 0.0604 - acc: 0.9757

32640/60000 [===============>..............] - ETA: 2s - loss: 0.0605 - acc: 0.9757

33280/60000 [===============>..............] - ETA: 2s - loss: 0.0605 - acc: 0.9756

33920/60000 [===============>..............] - ETA: 2s - loss: 0.0607 - acc: 0.9756

34560/60000 [================>.............] - ETA: 2s - loss: 0.0606 - acc: 0.9757

35072/60000 [================>.............] - ETA: 2s - loss: 0.0606 - acc: 0.9756

35712/60000 [================>.............] - ETA: 2s - loss: 0.0606 - acc: 0.9757

36352/60000 [=================>............] - ETA: 2s - loss: 0.0607 - acc: 0.9757

36992/60000 [=================>............] - ETA: 2s - loss: 0.0607 - acc: 0.9756

37632/60000 [=================>............] - ETA: 2s - loss: 0.0604 - acc: 0.9758

38272/60000 [==================>...........] - ETA: 1s - loss: 0.0608 - acc: 0.9756

38912/60000 [==================>...........] - ETA: 1s - loss: 0.0607 - acc: 0.9757

39552/60000 [==================>...........] - ETA: 1s - loss: 0.0607 - acc: 0.9757

40192/60000 [===================>..........] - ETA: 1s - loss: 0.0606 - acc: 0.9757

40832/60000 [===================>..........] - ETA: 1s - loss: 0.0605 - acc: 0.9758

41472/60000 [===================>..........] - ETA: 1s - loss: 0.0605 - acc: 0.9758

42112/60000 [====================>.........] - ETA: 1s - loss: 0.0606 - acc: 0.9758

42752/60000 [====================>.........] - ETA: 1s - loss: 0.0606 - acc: 0.9758

43392/60000 [====================>.........] - ETA: 1s - loss: 0.0605 - acc: 0.9758

44032/60000 [=====================>........] - ETA: 1s - loss: 0.0607 - acc: 0.9758

44672/60000 [=====================>........] - ETA: 1s - loss: 0.0607 - acc: 0.9758

45312/60000 [=====================>........] - ETA: 1s - loss: 0.0607 - acc: 0.9758

45952/60000 [=====================>........] - ETA: 1s - loss: 0.0608 - acc: 0.9757

46592/60000 [======================>.......] - ETA: 1s - loss: 0.0609 - acc: 0.9757

47232/60000 [======================>.......] - ETA: 1s - loss: 0.0608 - acc: 0.9757

47872/60000 [======================>.......] - ETA: 1s - loss: 0.0608 - acc: 0.9757

48512/60000 [=======================>......] - ETA: 1s - loss: 0.0610 - acc: 0.9756

49152/60000 [=======================>......] - ETA: 0s - loss: 0.0609 - acc: 0.9757

49792/60000 [=======================>......] - ETA: 0s - loss: 0.0609 - acc: 0.9757

50432/60000 [========================>.....] - ETA: 0s - loss: 0.0608 - acc: 0.9757

51072/60000 [========================>.....] - ETA: 0s - loss: 0.0608 - acc: 0.9757

51712/60000 [========================>.....] - ETA: 0s - loss: 0.0608 - acc: 0.9757

52352/60000 [=========================>....] - ETA: 0s - loss: 0.0607 - acc: 0.9758

52992/60000 [=========================>....] - ETA: 0s - loss: 0.0608 - acc: 0.9757

53632/60000 [=========================>....] - ETA: 0s - loss: 0.0609 - acc: 0.9757

54144/60000 [==========================>...] - ETA: 0s - loss: 0.0608 - acc: 0.9757

54784/60000 [==========================>...] - ETA: 0s - loss: 0.0608 - acc: 0.9757

55424/60000 [==========================>...] - ETA: 0s - loss: 0.0608 - acc: 0.9757

56064/60000 [===========================>..] - ETA: 0s - loss: 0.0606 - acc: 0.9758

56704/60000 [===========================>..] - ETA: 0s - loss: 0.0607 - acc: 0.9758

57344/60000 [===========================>..] - ETA: 0s - loss: 0.0607 - acc: 0.9758

57856/60000 [===========================>..] - ETA: 0s - loss: 0.0607 - acc: 0.9758

58496/60000 [============================>.] - ETA: 0s - loss: 0.0608 - acc: 0.9757

59136/60000 [============================>.] - ETA: 0s - loss: 0.0607 - acc: 0.9758

59776/60000 [============================>.] - ETA: 0s - loss: 0.0606 - acc: 0.9758



60000/60000 [==============================] - 5s 91us/step - loss: 0.0606 - acc: 0.9758


Epoch 4/5
  128/60000 [..............................] - ETA: 6s - loss: 0.0559 - acc: 0.9797

  640/60000 [..............................] - ETA: 6s - loss: 0.0603 - acc: 0.9761

 1280/60000 [..............................] - ETA: 5s - loss: 0.0600 - acc: 0.9765



 1920/60000 [..............................] - ETA: 5s - loss: 0.0599 - acc: 0.9768

 2560/60000 [>.............................] - ETA: 5s - loss: 0.0602 - acc: 0.9763

 3072/60000 [>.............................] - ETA: 5s - loss: 0.0598 - acc: 0.9762

 3712/60000 [>.............................] - ETA: 5s - loss: 0.0590 - acc: 0.9762

 4352/60000 [=>............................] - ETA: 5s - loss: 0.0582 - acc: 0.9767

 4992/60000 [=>............................] - ETA: 5s - loss: 0.0588 - acc: 0.9763

 5632/60000 [=>............................] - ETA: 5s - loss: 0.0592 - acc: 0.9764

 6272/60000 [==>...........................] - ETA: 5s - loss: 0.0591 - acc: 0.9763

 6784/60000 [==>...........................] - ETA: 5s - loss: 0.0588 - acc: 0.9766

 7296/60000 [==>...........................] - ETA: 5s - loss: 0.0583 - acc: 0.9766

 7936/60000 [==>...........................] - ETA: 4s - loss: 0.0577 - acc: 0.9768

 8576/60000 [===>..........................] - ETA: 4s - loss: 0.0578 - acc: 0.9767

 9216/60000 [===>..........................]

 - ETA: 4s - loss: 0.0581 - acc: 0.9766

 9856/60000 [===>..........................] - ETA: 4s - loss: 0.0579 - acc: 0.9766

10496/60000 [====>.........................] - ETA: 4s - loss: 0.0574 - acc: 0.9768



11136/60000 [====>.........................] - ETA: 4s - loss: 0.0571 - acc: 0.9768



11776/60000 [====>.........................] - ETA: 4s - loss: 0.0573 - acc: 0.9768

12416/60000 [=====>........................] - ETA: 4s - loss: 0.0577 - acc: 0.9765

13056/60000 [=====>........................] - ETA: 4s - loss: 0.0574 - acc: 0.9767

13568/60000 [=====>........................] - ETA: 4s - loss: 0.0577 - acc: 0.9766

14208/60000 [======>.......................] - ETA: 4s - loss: 0.0576 - acc: 0.9767

14848/60000 [======>.......................] - ETA: 4s - loss: 0.0576 - acc: 0.9767



15488/60000 [======>.......................] - ETA: 4s - loss: 0.0576 - acc: 0.9767

16000/60000 [=======>......................] - ETA: 4s - loss: 0.0574 - acc: 0.9769

16640/60000 [=======>......................] - ETA: 4s - loss: 0.0572 - acc: 0.9768

17280/60000 [=======>......................] - ETA: 3s - loss: 0.0569 - acc: 0.9769

17920/60000 [=======>......................] - ETA: 3s - loss: 0.0568 - acc: 0.9770

18560/60000 [========>.....................] - ETA: 3s - loss: 0.0565 - acc: 0.9771

19200/60000 [========>.....................] - ETA: 3s - loss: 0.0567 - acc: 0.9771

19840/60000 [========>.....................] - ETA: 3s - loss: 0.0566 - acc: 0.9770

20480/60000 [=========>....................] - ETA: 3s - loss: 0.0570 - acc: 0.9769



21120/60000 [=========>....................] - ETA: 3s - loss: 0.0571 - acc: 0.9769

21760/60000 [=========>....................] - ETA: 3s - loss: 0.0567 - acc: 0.9770

22400/60000 [==========>...................] - ETA: 3s - loss: 0.0570 - acc: 0.9769

23040/60000 [==========>...................] - ETA: 3s - loss: 0.0569 - acc: 0.9769

23680/60000 [==========>...................] - ETA: 3s - loss: 0.0567 - acc: 0.9769

24320/60000 [===========>..................] - ETA: 3s - loss: 0.0569 - acc: 0.9769

24960/60000 [===========>..................] - ETA: 3s - loss: 0.0569 - acc: 0.9769



25600/60000 [===========>..................] - ETA: 3s - loss: 0.0566 - acc: 0.9769

26240/60000 [============>.................] - ETA: 3s - loss: 0.0567 - acc: 0.9768

26880/60000 [============>.................] - ETA: 3s - loss: 0.0565 - acc: 0.9769

27520/60000 [============>.................] - ETA: 3s - loss: 0.0563 - acc: 0.9770

28160/60000 [=============>................] - ETA: 2s - loss: 0.0562 - acc: 0.9771

28800/60000 [=============>................] - ETA: 2s - loss: 0.0564 - acc: 0.9770

29440/60000 [=============>................] - ETA: 2s - loss: 0.0564 - acc: 0.9770

30080/60000 [==============>...............] - ETA: 2s - loss: 0.0565 - acc: 0.9770

30720/60000 [==============>...............] - ETA: 2s - loss: 0.0564 - acc: 0.9770

31232/60000 [==============>...............] - ETA: 2s - loss: 0.0563 - acc: 0.9771

31872/60000 [==============>...............] - ETA: 2s - loss: 0.0561 - acc: 0.9772

32512/60000 [===============>..............] - ETA: 2s - loss: 0.0561 - acc: 0.9772

33152/60000 [===============>..............] - ETA: 2s - loss: 0.0561 - acc: 0.9772



33792/60000 [===============>..............] - ETA: 2s - loss: 0.0560 - acc: 0.9773

34304/60000 [================>.............] - ETA: 2s - loss: 0.0561 - acc: 0.9773

34944/60000 [================>.............] - ETA: 2s - loss: 0.0561 - acc: 0.9772

35584/60000 [================>.............] - ETA: 2s - loss: 0.0562 - acc: 0.9773

36224/60000 [=================>............] - ETA: 2s - loss: 0.0562 - acc: 0.9773

36864/60000 [=================>............] - ETA: 2s - loss: 0.0561 - acc: 0.9773

37504/60000 [=================>............] - ETA: 2s - loss: 0.0560 - acc: 0.9773

38144/60000 [==================>...........] - ETA: 2s - loss: 0.0562 - acc: 0.9772

38784/60000 [==================>...........] - ETA: 1s - loss: 0.0563 - acc: 0.9772

39424/60000 [==================>...........] - ETA: 1s - loss: 0.0562 - acc: 0.9772

40064/60000 [===================>..........] - ETA: 1s - loss: 0.0562 - acc: 0.9772

40704/60000 [===================>..........] - ETA: 1s - loss: 0.0562 - acc: 0.9772

41344/60000 [===================>..........] - ETA: 1s - loss: 0.0561 - acc: 0.9772

41984/60000 [===================>..........] - ETA: 1s - loss: 0.0561 - acc: 0.9772

42624/60000 [====================>.........] - ETA: 1s - loss: 0.0562 - acc: 0.9772

43136/60000 [====================>.........] - ETA: 1s - loss: 0.0563 - acc: 0.9772

43776/60000 [====================>.........] - ETA: 1s - loss: 0.0563 - acc: 0.9771

44416/60000 [=====================>........] - ETA: 1s - loss: 0.0565 - acc: 0.9771

45056/60000 [=====================>........] - ETA: 1s - loss: 0.0565 - acc: 0.9771

45696/60000 [=====================>........] - ETA: 1s - loss: 0.0564 - acc: 0.9771

46336/60000 [======================>.......] - ETA: 1s - loss: 0.0563 - acc: 0.9772

46976/60000 [======================>.......] - ETA: 1s - loss: 0.0565 - acc: 0.9771

47616/60000 [======================>.......] - ETA: 1s - loss: 0.0565 - acc: 0.9772

48128/60000 [=======================>......] - ETA: 1s - loss: 0.0565 - acc: 0.9771

48768/60000 [=======================>......] - ETA: 1s - loss: 0.0565 - acc: 0.9771

49408/60000 [=======================>......] - ETA: 0s - loss: 0.0565 - acc: 0.9771

50048/60000 [========================>.....] - ETA: 0s - loss: 0.0564 - acc: 0.9772

50688/60000 [========================>.....] - ETA: 0s - loss: 0.0563 - acc: 0.9773

51328/60000 [========================>.....] - ETA: 0s - loss: 0.0562 - acc: 0.9773

51968/60000 [========================>.....] - ETA: 0s - loss: 0.0562 - acc: 0.9773

52608/60000 [=========================>....] - ETA: 0s - loss: 0.0563 - acc: 0.9773

53248/60000 [=========================>....] - ETA: 0s - loss: 0.0562 - acc: 0.9774



53888/60000 [=========================>....] - ETA: 0s - loss: 0.0563 - acc: 0.9773

54528/60000 [==========================>...] - ETA: 0s - loss: 0.0564 - acc: 0.9773

55168/60000 [==========================>...] - ETA: 0s - loss: 0.0564 - acc: 0.9773

55808/60000 [==========================>...] - ETA: 0s - loss: 0.0564 - acc: 0.9773

56448/60000 [===========================>..] - ETA: 0s - loss: 0.0564 - acc: 0.9773

57088/60000 [===========================>..] - ETA: 0s - loss: 0.0564 - acc: 0.9773

57728/60000 [===========================>..] - ETA: 0s - loss: 0.0564 - acc: 0.9773

58368/60000 [============================>.] - ETA: 0s - loss: 0.0563 - acc: 0.9773

59008/60000 [============================>.] - ETA: 0s - loss: 0.0563 - acc: 0.9773

59648/60000 [============================>.] - ETA: 0s - loss: 0.0564 - acc: 0.9773

60000/60000 [==============================] - 5s 92us/step - loss: 0.0565 - acc: 0.9773


Epoch 5/5
  128/60000 [..............................] - ETA: 6s - loss: 0.0677 - acc: 0.9727

  640/60000 [..............................] - ETA: 6s - loss: 0.0503 - acc: 0.9798

 1280/60000 [..............................] - ETA: 5s - loss: 0.0518 - acc: 0.9795



 1920/60000 [..............................] - ETA: 5s - loss: 0.0516 - acc: 0.9792

 2432/60000 [>.............................] - ETA: 5s - loss: 0.0513 - acc: 0.9793

 2944/60000 [>.............................] - ETA: 5s - loss: 0.0524 - acc: 0.9791

 3584/60000 [>.............................] - ETA: 5s - loss: 0.0551 - acc: 0.9779

 4224/60000 [=>............................] - ETA: 5s - loss: 0.0550 - acc: 0.9781

 4864/60000 [=>............................] - ETA: 5s - loss: 0.0552 - acc: 0.9783

 5504/60000 [=>............................] - ETA: 5s - loss: 0.0567 - acc: 0.9776



 6144/60000 [==>...........................] - ETA: 5s - loss: 0.0560 - acc: 0.9780



 6784/60000 [==>...........................] - ETA: 5s - loss: 0.0555 - acc: 0.9782

 7424/60000 [==>...........................] - ETA: 5s - loss: 0.0550 - acc: 0.9784

 8064/60000 [===>..........................] - ETA: 4s - loss: 0.0545 - acc: 0.9785

 8704/60000 [===>..........................] - ETA: 4s - loss: 0.0550 - acc: 0.9785

 9344/60000 [===>..........................] - ETA: 4s - loss: 0.0541 - acc: 0.9789

 9984/60000 [===>..........................] - ETA: 4s - loss: 0.0537 - acc: 0.9789

10624/60000 [====>.........................] - ETA: 4s - loss: 0.0539 - acc: 0.9788

11264/60000 [====>.........................] - ETA: 4s - loss: 0.0534 - acc: 0.9790

11904/60000 [====>.........................] - ETA: 4s - loss: 0.0533 - acc: 0.9789

12544/60000 [=====>........................] - ETA: 4s - loss: 0.0529 - acc: 0.9790



13184/60000 [=====>........................] - ETA: 4s - loss: 0.0526 - acc: 0.9790

13824/60000 [=====>........................] - ETA: 4s - loss: 0.0533 - acc: 0.9788

14464/60000 [======>.......................] - ETA: 4s - loss: 0.0533 - acc: 0.9788

15104/60000 [======>.......................] - ETA: 4s - loss: 0.0533 - acc: 0.9787

15744/60000 [======>.......................] - ETA: 4s - loss: 0.0534 - acc: 0.9785

16384/60000 [=======>......................] - ETA: 4s - loss: 0.0533 - acc: 0.9786

17024/60000 [=======>......................] - ETA: 3s - loss: 0.0535 - acc: 0.9786

17664/60000 [=======>......................] - ETA: 3s - loss: 0.0536 - acc: 0.9786

18304/60000 [========>.....................] - ETA: 3s - loss: 0.0537 - acc: 0.9786

18944/60000 [========>.....................] - ETA: 3s - loss: 0.0534 - acc: 0.9786

19584/60000 [========>.....................] - ETA: 3s - loss: 0.0535 - acc: 0.9787

20224/60000 [=========>....................] - ETA: 3s - loss: 0.0533 - acc: 0.9787

20864/60000 [=========>....................] - ETA: 3s - loss: 0.0534 - acc: 0.9787



21504/60000 [=========>....................] - ETA: 3s - loss: 0.0531 - acc: 0.9788

22016/60000 [==========>...................] - ETA: 3s - loss: 0.0532 - acc: 0.9788

22528/60000 [==========>...................] - ETA: 3s - loss: 0.0531 - acc: 0.9788

23040/60000 [==========>...................] - ETA: 3s - loss: 0.0530 - acc: 0.9788

23552/60000 [==========>...................] - ETA: 3s - loss: 0.0529 - acc: 0.9788

24192/60000 [===========>..................] - ETA: 3s - loss: 0.0532 - acc: 0.9787

24832/60000 [===========>..................] - ETA: 3s - loss: 0.0533 - acc: 0.9787

25472/60000 [===========>..................] - ETA: 3s - loss: 0.0534 - acc: 0.9787

26112/60000 [============>.................] - ETA: 3s - loss: 0.0534 - acc: 0.9787

26752/60000 [============>.................] - ETA: 3s - loss: 0.0533 - acc: 0.9787

27392/60000 [============>.................] - ETA: 3s - loss: 0.0535 - acc: 0.9787

28032/60000 [=============>................] - ETA: 2s - loss: 0.0535 - acc: 0.9787

28544/60000 [=============>................] - ETA: 2s - loss: 0.0535 - acc: 0.9787

29056/60000 [=============>................] - ETA: 2s - loss: 0.0534 - acc: 0.9787

29568/60000 [=============>................] - ETA: 2s - loss: 0.0534 - acc: 0.9788

30208/60000 [==============>...............] - ETA: 2s - loss: 0.0533 - acc: 0.9788

30720/60000 [==============>...............] - ETA: 2s - loss: 0.0532 - acc: 0.9789

31360/60000 [==============>...............] - ETA: 2s - loss: 0.0531 - acc: 0.9789

32000/60000 [===============>..............] - ETA: 2s - loss: 0.0530 - acc: 0.9789

32512/60000 [===============>..............] - ETA: 2s - loss: 0.0529 - acc: 0.9789

33024/60000 [===============>..............] - ETA: 2s - loss: 0.0532 - acc: 0.9789

33536/60000 [===============>..............] - ETA: 2s - loss: 0.0531 - acc: 0.9789

34176/60000 [================>.............] - ETA: 2s - loss: 0.0531 - acc: 0.9788

34688/60000 [================>.............] - ETA: 2s - loss: 0.0531 - acc: 0.9788

35200/60000 [================>.............] - ETA: 2s - loss: 0.0530 - acc: 0.9789

35712/60000 [================>.............] - ETA: 2s - loss: 0.0530 - acc: 0.9789

36224/60000 [=================>............] - ETA: 2s - loss: 0.0530 - acc: 0.9789

36736/60000 [=================>............] - ETA: 2s - loss: 0.0531 - acc: 0.9788

37248/60000 [=================>............] - ETA: 2s - loss: 0.0533 - acc: 0.9787

37760/60000 [=================>............] - ETA: 2s - loss: 0.0533 - acc: 0.9788

38272/60000 [==================>...........] - ETA: 2s - loss: 0.0533 - acc: 0.9788

38784/60000 [==================>...........] - ETA: 2s - loss: 0.0534 - acc: 0.9787

39296/60000 [==================>...........] - ETA: 1s - loss: 0.0534 - acc: 0.9788

39808/60000 [==================>...........] - ETA: 1s - loss: 0.0534 - acc: 0.9787

40320/60000 [===================>..........] - ETA: 1s - loss: 0.0534 - acc: 0.9788

40832/60000 [===================>..........] - ETA: 1s - loss: 0.0535 - acc: 0.9787

41344/60000 [===================>..........] - ETA: 1s - loss: 0.0535 - acc: 0.9787

41984/60000 [===================>..........] - ETA: 1s - loss: 0.0536 - acc: 0.9787

42496/60000 [====================>.........] - ETA: 1s - loss: 0.0536 - acc: 0.9787

43008/60000 [====================>.........] - ETA: 1s - loss: 0.0536 - acc: 0.9787

43392/60000 [====================>.........] - ETA: 1s - loss: 0.0535 - acc: 0.9787

44032/60000 [=====================>........] - ETA: 1s - loss: 0.0534 - acc: 0.9788

44544/60000 [=====================>........] - ETA: 1s - loss: 0.0534 - acc: 0.9788

45056/60000 [=====================>........] - ETA: 1s - loss: 0.0534 - acc: 0.9788

45568/60000 [=====================>........] - ETA: 1s - loss: 0.0534 - acc: 0.9787

46080/60000 [======================>.......] - ETA: 1s - loss: 0.0535 - acc: 0.9787

46464/60000 [======================>.......] - ETA: 1s - loss: 0.0534 - acc: 0.9787

46976/60000 [======================>.......] - ETA: 1s - loss: 0.0533 - acc: 0.9787

47488/60000 [======================>.......] - ETA: 1s - loss: 0.0534 - acc: 0.9787

48000/60000 [=======================>......] - ETA: 1s - loss: 0.0534 - acc: 0.9787

48512/60000 [=======================>......] - ETA: 1s - loss: 0.0535 - acc: 0.9787

49024/60000 [=======================>......] - ETA: 1s - loss: 0.0535 - acc: 0.9787

49536/60000 [=======================>......] - ETA: 1s - loss: 0.0535 - acc: 0.9787

50048/60000 [========================>.....] - ETA: 0s - loss: 0.0535 - acc: 0.9787

50560/60000 [========================>.....] - ETA: 0s - loss: 0.0535 - acc: 0.9787

51072/60000 [========================>.....] - ETA: 0s - loss: 0.0535 - acc: 0.9787

51584/60000 [========================>.....] - ETA: 0s - loss: 0.0534 - acc: 0.9788

52096/60000 [=========================>....] - ETA: 0s - loss: 0.0534 - acc: 0.9788

52736/60000 [=========================>....] - ETA: 0s - loss: 0.0534 - acc: 0.9788

53248/60000 [=========================>....] - ETA: 0s - loss: 0.0533 - acc: 0.9788

53760/60000 [=========================>....] - ETA: 0s - loss: 0.0533 - acc: 0.9788

54272/60000 [==========================>...] - ETA: 0s - loss: 0.0534 - acc: 0.9787

54784/60000 [==========================>...] - ETA: 0s - loss: 0.0534 - acc: 0.9787

55296/60000 [==========================>...] - ETA: 0s - loss: 0.0535 - acc: 0.9787

55808/60000 [==========================>...] - ETA: 0s - loss: 0.0535 - acc: 0.9787



56448/60000 [===========================>..] - ETA: 0s - loss: 0.0535 - acc: 0.9787

57088/60000 [===========================>..] - ETA: 0s - loss: 0.0535 - acc: 0.9787



57728/60000 [===========================>..] - ETA: 0s - loss: 0.0534 - acc: 0.9787

58368/60000 [============================>.] - ETA: 0s - loss: 0.0535 - acc: 0.9787

59008/60000 [============================>.] - ETA: 0s - loss: 0.0535 - acc: 0.9787

59520/60000 [============================>.] - ETA: 0s - loss: 0.0535 - acc: 0.9787

60000/60000 [==============================] - 6s 98us/step - loss: 0.0534 - acc: 0.9788


   32/10000 [..............................] - ETA: 6s

 1312/10000 [==>...........................] - ETA: 0s

 2624/10000 [======>.......................] - ETA: 0s



 3936/10000 [==========>...................] - ETA: 0s

 5184/10000 [==============>...............] - ETA: 0s

 6496/10000 [==================>...........] - ETA: 0s

 7616/10000 [=====================>........] - ETA: 0s

 8928/10000 [=========================>....] - ETA: 0s

10000/10000 [==============================] - 0s 43us/step


Test Accuracy:  0.9761900050163269


Using the binary_crossentropy loss metric was the primary tool I used to enhance accuracy. Because it uses a log_loss function, it highly punishes guesses that are far from the actual label. Because of this, we almost run the risk of overfitting, but still have an Accuracy of 97% on our testing data. 